internal/compiler/generator.rs

```diff
@@ -315,25 +315,31 @@ pub fn build_item_tree<T: ItemTreeBuilder>(
         parent_index: u32,
         builder: &mut T,
     ) {
-        if item.borrow().repeated.is_some() {
-            builder.push_repeated_item(item, *repeater_count, parent_index, component_state);
-            *repeater_count += 1;
-        } else {
-            let mut item = item.clone();
-            let mut component_state = component_state.clone();
-            while let Some((base, state)) = {
-                let base = item.borrow().sub_component().map(|c| {
-                    (
-                        c.root_element.clone(),
-                        builder.enter_component(&item, c, children_offset, &component_state),
-                    )
-                });
-                base
-            } {
-                item = base;
-                component_state = state;
+        match &item.borrow().repeated {
+            Some(RepeatedElementInfo::Repeater(_)) => {
+                builder.push_repeated_item(item, *repeater_count, parent_index, component_state);
+                *repeater_count += 1;
+            }
+            Some(RepeatedElementInfo::Embedding(_)) => {
+                todo!();
```

Comment:
here you may want indeed to push some different node in the item for the dynamic node.